s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3SourceSpec.scala

```diff
@@ -17,14 +18,28 @@ class S3SourceSpec extends S3WireMockBase with S3ClientIntegrationSpec {
     mockDownload()
 
     //#download
-    val s3Source: Source[ByteString, NotUsed] = s3Client.download("testBucket", "testKey")
+    val s3Source: Source[ByteString, NotUsed] = s3Client.download(bucket, bucketKey)
     //#download
 
     val result: Future[String] = s3Source.map(_.utf8String).runWith(Sink.head)
 
     result.futureValue shouldBe body
   }
 
+  it should "download a range of file's bytes from S3 if bytes range given" in {
+
+    mockRangedDownload()
+
+    //#rangedDownload
+    val s3Source: Source[ByteString, NotUsed] =
+      s3Client.download(bucket, bucketKeyForRanged, ByteRange(bytesRangeStart, bytesRangeEnd))
+    //#rangedDownload
+
+    val result: Future[Array[Byte]] = s3Source.map(_.toArray).runWith(Sink.head)
```

Comment:
minor: this could be consistent with the previous tests and deal with strings (doing `map(_.utf8string)`).
this will need
`val rangeOfBody = body.substring(bytesRangeStart, bytesRangeEnd + 1)`